### PR TITLE
Clean timeout tablets channel in TabletWriterMgr

### PR DESCRIPTION
**Pull request by kangkaisen**
Fix https://github.com/apache/incubator-doris/issues/690

Add a config `streaming_load_max_duration_time_sec` , a thread `_tablets_channel_clean_thread` and a `std::unordered_map<TabletsChannelKey, time_t,TabletsChannelKeyHasher> _key_time_map`

If a tablets_channel alive time exceed the `streaming_load_max_duration_time_sec`, we will erase the tablets_channel.

This PR is testing because the test need a long time.

### CODE REVIEW

File: be/src/runtime/tablet_writer_mgr.h [hunk 1]
```diff
@@ -92,6 +96,15 @@ class TabletWriterMgr {
         TabletsChannelKeyHasher> _tablets_channels;
 
     Cache* _lastest_success_channel = nullptr;
+
+    std::unordered_map<TabletsChannelKey, time_t,TabletsChannelKeyHasher> _key_time_map;
```
Comment (imay) [inline be/src/runtime/tablet_writer_mgr.h]: I think it's better to add a `last_updated_time` member in `TabletsChannel `, then there is no need to maintain two map.

And when there is a new request referring to this tablets, then update this field.

Then we can compare current wall time to this `last_updated_time` to see if it has been exceeded the config timeout.

And I think this timeout can be some shorter time, such as 10 minutes

File: be/src/runtime/tablet_writer_mgr.cpp [hunk 2]
```diff
@@ -311,6 +313,44 @@ Status TabletWriterMgr::cancel(const PTabletWriterCancelRequest& params) {
...
         _tablets_channels.erase(key);
+        _key_time_map.erase(key);
+    }
+    return Status::OK;
+}
+
+Status TabletWriterMgr::start_bg_worker() {
+    _tablets_channel_clean_thread = std::thread([this] {
+                _tablets_channel_clean_thread_callback(nullptr);
+            });
```
Comment (imay) [inline be/src/runtime/tablet_writer_mgr.cpp]: you need to call `_tablets_channel_clean_thread.detach()` to avoid crush when process exits.

File: be/src/runtime/tablet_writer_mgr.cpp [hunk 3]
```diff
@@ -311,6 +313,44 @@ Status TabletWriterMgr::cancel(const PTabletWriterCancelRequest& params) {
...
         std::lock_guard<std::mutex> l(_lock);
         _tablets_channels.erase(key);
+        _key_time_map.erase(key);
+    }
+    return Status::OK;
+}
+
+Status TabletWriterMgr::start_bg_worker() {
+    _tablets_channel_clean_thread = std::thread([this] {
+                _tablets_channel_clean_thread_callback(nullptr);
```
Comment (imay) [inline be/src/runtime/tablet_writer_mgr.cpp]:
```suggestion
        uint32_t interval = 60;
        while (true) {
            _start_tablets_channel_clean();
            sleep(interval);
        }
```
I think 600 is a bit long

**Review by imay — CHANGES_REQUESTED** [review]

File: be/src/runtime/tablet_writer_mgr.h (hunk 1 above)
Comment (kangkaisen) [reply be/src/runtime/tablet_writer_mgr.h]: OK

File: be/src/runtime/tablet_writer_mgr.cpp (hunk 2 above)
Comment (kangkaisen) [reply be/src/runtime/tablet_writer_mgr.cpp]: OK

File: be/src/runtime/tablet_writer_mgr.cpp (hunk 3 above)
Comment (kangkaisen) [reply be/src/runtime/tablet_writer_mgr.cpp]: OK

File: be/src/runtime/tablet_writer_mgr.cpp [hunk 4]
```diff
@@ -315,6 +324,48 @@ Status TabletWriterMgr::cancel(const PTabletWriterCancelRequest& params) {
...
+Status TabletWriterMgr::start_bg_worker() {
+    _tablets_channel_clean_thread = std::thread(
+         [this] {
+             _tablets_channel_clean_thread_callback(nullptr);
+         });
+    _tablets_channel_clean_thread.detach();
+    return Status::OK;
+}
+
+void* TabletWriterMgr::_tablets_channel_clean_thread_callback(void* arg) {
```
Comment (imay) [inline be/src/runtime/tablet_writer_mgr.cpp]: you can remove this function and put this logical to lambda in `start_bg_worker`.

Comment (kangkaisen) [reply be/src/runtime/tablet_writer_mgr.cpp]: done

File: be/src/runtime/tablet_writer_mgr.cpp [hunk 5]
```diff
@@ -315,6 +324,45 @@ Status TabletWriterMgr::cancel(const PTabletWriterCancelRequest& params) {
...
+        std::vector<TabletsChannelKey> need_delete_keys;
+
+        for (auto& kv : _tablets_channels) {
+            time_t last_updated_time = kv.second->last_updated_time();
+            if (difftime(now, last_updated_time) >= max_alive_time) {
+                need_delete_keys.emplace_back(kv.first);
+            }
+        }
+
+        for(auto key: need_delete_keys) {
```
Comment (imay) [inline be/src/runtime/tablet_writer_mgr.cpp]:
```suggestion
        for(auto& key: need_delete_keys) {
```

**Review by imay — APPROVED** [review]
